### ISSUE TITLE
Using TableName and AutoFilter - Not compatible 

### ISSUE THREAD

**Issue by Penkethboy** [dfinke/ImportExcel#353]
While testing adding -Tablename <name> to my existing code i received the following error

Export-Excel : Parameter set cannot be resolved using the specified named parameters.
At c:\test\testing.ps1:600 char:31
+ ... oviesData | Export-Excel $EXcelFile -WorkSheetname $SheetName -TableN ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Export-Excel], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousParameterSet,Export-Excel

This as it turned out was because in my code i was also using -AutoFilter to generate the Excel file

I understand after a bit of testing that -TableName also applies an Autofilter so my code was trying to apply it twice.

Maybe Export-Excel should "ignore" and/or warn about incompatible parameters?

It would be good if the error message gave a bit more informative text as to the likely issue - rather than a generic error - so you don't have to feel the force to understand it :)

**Comment by jhoneill** [comment]
This is by design. 
AutoFilter is basically a table underneath, so you can't create a table and an autofilter on the same set of cells. 
Export-Excel has it's parameters designed so you can not specify both in the same command. If you put one in intellisense won't offer you the other. If you create a command line with both you get a message which says "None of the parameter sets have this combination of parameters". 

To avoid an unworkable number of parameter sets when I added append, I put in something in the body to throw an error if you say you want to append to the sheet and to clear it. The error is nicer that way.

**Comment by Penkethboy** [comment]
It may be by design but it does not work as you say
I don't get the error message when using VS Code
![capture1a](https://user-images.githubusercontent.com/21155304/41821158-60471e5e-77d4-11e8-989f-ea42aefa9b22.JPG)
and it runs fine until export-excel is initiated

Not sure what your append sentence is about?

**Comment by jhoneill** [comment]
AFAIK, VS Code currently offers all the parameter sets - hence the 1/5 
The 5 parameter sets are "Default" (file and AutoFilter), "Package Default" (excel package object and AutoFilter) "Table" (file and named table), "PackageTable" (excel package object and named table), and "now". 
 Intellisense in the ISE or the normal shell spots when the parameters supplied so far mean that that you must be using a parameter set which doesn't allow some other parameter and cuts down the list of parameters. VS Code doesn't.

So I ended up with 4 parameter sets for file/object and Named-Table/AutoRange; if I kept going with that I would have 4 sets which allowed removed the things which can't be done when appending (e.g. Clear sheet) and 4 which didn't.  And the next parameter which excluded something would need 8 excluding sets and 8 including sets. Eventually you get to the point of saying "I'm not going to double the number of sets just so the ISE prunes out clearsheet or append when the other is selected" and just put some validation code in which throws an error if someone provides both.
